internal/pkg/platform/schema/compiler/extension/primarykey/mixin.go

```diff
@@ -0,0 +1,151 @@
+package primarykey
+
+import (
+	"reflect"
+
+	"entgo.io/ent"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/strhelper"
+)
+
+const ParentEdgeName = "parent"
+
+type Mixin struct {
+	ent.Schema
+	pkConfig
+}
+
+// Option used with the Definition function.
+type Option func(*pkConfig)
+
+type pkConfig struct {
+	parent      ent.Interface
+	parentMixin *Mixin
+	pkFields    []pkFieldConfig
+}
+
+type pkFieldConfig struct {
+	Name string
+	Type any
+}
+
+// WithChildOf can be used at most once in a call of the Mixin Definition function.
+// All primary keys of all parents will be included in the target primary key.
+func WithChildOf(parent ent.Interface) Option {
+	return func(config *pkConfig) {
+		if config.parent != nil {
+			panic(errors.Errorf(`primarykey.ChildOf option can be used at most once`))
+		}
+		config.parent = parent
+	}
+}
+
+// WithField must be used at least once in a call of the Mixin Definition function.
+// This option adds the field to the target primary key.
+// Types can be any type that is based on string or int types.
+func WithField(name string, typ any) Option {
+	return func(config *pkConfig) {
+		name = strhelper.FirstLower(name)
+		config.pkFields = append(config.pkFields, pkFieldConfig{Name: name, Type: typ})
+	}
+}
+
+// Definition creates primary key definition as an ent mixin.
+// Read more: https://entgo.io/docs/schema-mixin
+func Definition(ops ...Option) ent.Mixin {
+	// Apply options
+	c := pkConfig{}
+	for _, o := range ops {
+		o(&c)
+	}
+
+	// At lest one field is required
+	if len(c.pkFields) == 0 {
+		panic(errors.Errorf(`at least one field must be specified, please use primarykey.Field(name, type) option`))
+	}
+
+	// Merge PKs from the parent, if any
+	if c.parent != nil {
+		for _, mixin := range c.parent.Mixin() {
+			if m, ok := mixin.(*Mixin); ok {
+				c.parentMixin = m
+				c.pkFields = append(m.pkConfig.pkFields, c.pkFields...)
+			}
+		}
+	}
+
+	return &Mixin{pkConfig: c}
+}
+
+// Fields method generates primary key partial fields.
+func (v *Mixin) Fields() []ent.Field {
+	var fields []ent.Field
+
+	// Mark the ID field with the annotation and define GoType.
+	// At the time of loading of this definition, there is no generated structure for the primary key yet.
+	// So the KeyPlaceholder is used (it has same methods), and it is replaced by the generated struct in the Extension.Hooks.
+	fields = append(fields, field.String("id").NotEmpty().Immutable().GoType(KeyPlaceholder{}).Annotations(v.annotation()))
```

Comment:
V case citania tejto definicie este nie su vygenerovane typy pre `ID` fields (`BranchKey`,`ConfigurationKey`, ....).

Preto tu pouzivam tuto "fintu".

Ako typ sa nastavi `KeyPlaceholder`, ktory ma rovnake metody ako budu mat vygenerovane kluce.
A neskor sa v pomocou `extension` nahradi definicia typu, za ten vygenerovany.

Su tam nejake technicke dovody, preco sa to musi spravit takto.

V skrate: `entgo` je stavane tak, aby sa dalo na zaklade schemy vygenerovat hocico, ale nerata sa s tym, ze uz aj nejake casti schemy sa do-generuju, v tomto pripade tieto `keys structs`.